ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java

```diff
@@ -118,22 +133,88 @@ public void setArguments(Map<String, Object> arguments) {
     }
   }
 
+  public class CqlTransformRequest {
+    private List<CqlRequestImpl> searches;
+    private int count;
+    private List<CqlRequest.Sort> sorts;
+    private List<String> hiddenResults;
+
+    public CqlTransformRequest() {
+      this.searches = Collections.emptyList();
+      this.count = 0;
+      this.sorts = Collections.emptyList();
+      this.hiddenResults = Collections.emptyList();
+    }
+
+    public void setSearches(List<CqlRequestImpl> searches) {
+      this.searches = searches;
+    }
+
+    public List<CqlRequestImpl> getSearches() {
+      return this.searches;
+    }
+
+    public void setCount(int count) {
+      this.count = count;
+    }
+
+    public int getCount() {
+      return this.count;
+    }
+
+    public void setSorts(List<CqlRequest.Sort> sorts) {
+      this.sorts = sorts;
+    }
+
+    public List<CqlRequest.Sort> getSorts() {
+      return this.sorts;
+    }
+
+    public List<CqlRequestImpl> getCqlRequests() {
+      this.searches.forEach(
+          cqlRequest -> {
+            cqlRequest.setSorts(this.sorts);
+          });
+      return this.searches;
+    }
+
+    public void setHiddenResults(List<String> hiddenResults) {
+      this.hiddenResults = hiddenResults;
+    }
+
+    public List<String> getHiddenResults() {
+      return this.hiddenResults;
+    }
+  }
+
   @Override
   public Object handle(Request request, Response response) throws Exception {
     String transformerId = request.params(":transformerId");
     String body = util.safeGetBody(request);
-    CqlRequestImpl cqlRequest;
+    CqlTransformRequest cqlTransformRequest;
 
     try {
-      cqlRequest = GSON.fromJson(body, CqlRequestImpl.class);
+      cqlTransformRequest = GSON.fromJson(body, CqlTransformRequest.class);
     } catch (IllegalArgumentException | ArrayIndexOutOfBoundsException e) {
       LOGGER.debug("Error fetching cql request");
       response.status(HttpStatus.BAD_REQUEST_400);
       return ImmutableMap.of("message", "Error retrieving cql request");
     }
 
-    if (cqlRequest.getCql() == null) {
+    List<CqlRequest> cqlRequests =
+        cqlTransformRequest
+            .getCqlRequests()
+            .stream()
+            .filter(
+                cqlRequest ->
+                    cqlRequest.getCql() != null
+                        && (cqlRequest.getSrc() != null
+                            || CollectionUtils.isNotEmpty(cqlRequest.getSrcs())))
```

Comment:
❓ Does `isNotEmpty` already check for `null`?

Reply:
It does, I'm checking the list of `srcs` in this line or for a single source in the line above